src/main/scala/nl.knaw.dans.easy.bagstore/BagStoreEnum.scala

```diff
@@ -22,22 +22,25 @@ import scala.util.Try
 
 trait BagStoreEnum { this: BagFacadeComponent with BagStoreContext =>
 
-  def enumBags(includeVisible: Boolean = true, includeHidden: Boolean = false): Try[Stream[BagId]] = Try {
-    Files.walk(baseDir, uuidPathComponentSizes.size).iterator().asScala.toStream
-      .map(baseDir.relativize)
-      .withFilter(_.getNameCount == uuidPathComponentSizes.size)
-      .map(p => fromLocation(baseDir.resolve(p)).flatMap(_.toBagId).get) // TODO: is there a better way to fail fast ?
-      .filter(bagId => {
+  // TODO: support huge numbers of bags. (The stream should then probably NOT be converted in to a List anymore!)
+  def enumBags(includeVisible: Boolean = true, includeHidden: Boolean = false): Try[Seq[BagId]] = Try {
+    resource.managed(Files.walk(baseDir, uuidPathComponentSizes.size)).acquireAndGet {
```

Comment:
I'd say that importing `resource._` in the top of this file and just calling `managed` here looks better, but that is just a matter of taste...